topics/releases.md

```diff
@@ -1,144 +1,109 @@
-branch, which had been branched off from `unstable` earlier.
+## Versioning
 
-Bug fixes and new features that can be stabilized during the release's time
-frame are committed to the unstable branch and backported to the release
-candidate branch. The `unstable` branch may include additional work that is not
-a part of the release candidate and scheduled for future releases.
+Valkey stable releases will generally follow `major.minor.patch` [semantic versioning schema](https://semver.org/).
+We follow semantic versioning to provide explicit guarantees regarding backward compatibility.
 
-The first release candidate, or RC1, is released once it can be used for
-development purposes and for testing the new version. At this stage, most of
-the new features and changes the new version brings are ready for review, and
-the release's purpose is collecting the public's feedback.
+When discussing compatibility, we refer to the following API contracts:
 
-Subsequent release candidates are released every three weeks or so, primarily
-for fixing bugs. These may also add new features and introduce changes, but at
-a decreasing rate and decreasing potential risk towards the final release
-candidate.
+1. Valkey commands including their inputs, outputs, and defined behavior
+2. The functions and APIs that can be executed from a Lua script
+3. The RDB version
+4. The protocol used to establish and replicate data from primaries to replicas
+5. The protocol between nodes within a Valkey cluster
+6. The Valkey Module API interface
 
-### Stable tree
+### Patch versions
 
-Once development has ended and the frequency of critical bug reports for the
-release candidate wanes, it is ready for the final release. At this point, the
-release is marked as stable and is released with "0" as its patch-level
-version.
+*Patch* versions are released with backwards compatible bug fixes and should not introduce new features.
 
-## Versioning
+Upgrading from a previous patch version should be safe and seamless.
+It should be safe to run a Valkey cluster with servers running on different patch versions.
 
-Stable releases liberally follow the usual `major.minor.patch` semantic
-versioning schema. The primary goal is to provide explicit guarantees regarding
-backward compatibility.
+*Patch* versions may also introduce small improvements such as performance or memory optimizations that are easy to verify as safe.
 
-### Patch-Level versions
+### Minor versions
 
-Patches primarily consist of bug fixes and very rarely introduce any
-compatibility issues.
+*Minor* versions are released with new functionality that is added in a backward compatible manner.
+Examples of new functionality include new commands, info fields, or configuration parameters.
 
-Upgrading from a previous patch-level version is almost always safe and
-seamless.
+Upgrading from a previous minor version should be safe, and will not introduce incompatibilities between servers in the cluster.
```

Comment:
```suggestion
Upgrading from a previous minor version should be safe, and will not introduce incompatibilities between servers in the cluster if default server configurations are not changed.
```

Reply:
I interpreted your suggestion as if you are using default configuration, behavior won't change. The current wording you have meant to me that the behavior won't change if **we** don't change default configurations, which we don't do in a minor version.